File: apps/android/core/designsystem/src/main/java/com/scribblefit/core/designsystem/components/ExerciseCardComponents.kt
package com.scribblefit.core.designsystem.components

import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.ColumnScope
import androidx.compose.foundation.layout.PaddingValues
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.rounded.History
import androidx.compose.material3.Icon
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.alpha
import androidx.compose.ui.draw.clip
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import androidx.compose.ui.res.stringResource
import com.scribblefit.core.designsystem.R
import com.scribblefit.core.designsystem.ScribbleFitTheme
import com.scribblefit.core.designsystem.scribbleGlass

@Composable
fun TrainingExerciseCard(
    name: String,
    formattedSummary: String,
    estimated1RM: String?,
    intensity: String?,
    improvement: String?,
    hasStats: Boolean,
    onClick: () -> Unit,
    modifier: Modifier = Modifier,
    alpha: Float = 1f,
    fontSize: Int = 24,
    kerning: Double = -0.5
) {
    GlassCard(
        modifier = modifier,
        onClick = onClick,
        alpha = alpha,
        contentPadding = PaddingValues(ScribbleFitTheme.spacing.extraSmall)
    ) {
        ExerciseHeader(
            name = name,
            formattedSummary = formattedSummary,
            fontSize = fontSize,
            kerning = kerning
        )
        ExerciseStats(
            estimated1RM = estimated1RM,
            intensity = intensity,
            improvement = improvement,
            hasStats = hasStats,
            estimated1RMLabel = stringResource(R.string.common_estimated_1rm),
            intensityLabel = stringResource(R.string.common_intensity)
        )
    }
}

@Composable
fun GlassCard(
    modifier: Modifier = Modifier,
    onClick: (() -> Unit)? = null,
    alpha: Float = 1f,
    contentPadding: PaddingValues = PaddingValues(ScribbleFitTheme.spacing.medium),
    content: @Composable ColumnScope.() -> Unit
) {
    Box(
        modifier = modifier
            .fillMaxWidth()
            .clip(RoundedCornerShape(ScribbleFitTheme.shapes.medium))
            .then(if (onClick != null) Modifier.clickable(onClick = onClick) else Modifier)
            .scribbleGlass(cornerRadius = ScribbleFitTheme.shapes.medium)
            .alpha(alpha)
    ) {
        Column(
            modifier = Modifier.padding(contentPadding),
            verticalArrangement = Arrangement.spacedBy(ScribbleFitTheme.spacing.medium),
            content = content
        )
    }
}

@Composable
fun ExerciseHeader(
    name: String,
    formattedSummary: String,
    modifier: Modifier = Modifier,
    fontSize: Int = 24,
    kerning: Double = -0.5
) {
    Column(
        modifier = modifier,
        verticalArrangement = Arrangement.spacedBy(ScribbleFitTheme.spacing.extraSmall)
    ) {
        Text(
            text = name,
            style = ScribbleFitTheme.typography.titleLarge.copy(
                fontSize = fontSize.sp,
                fontWeight = FontWeight.Bold,
                letterSpacing = kerning.sp
            ),
            color = ScribbleFitTheme.colors.primary
        )
        Text(
            text = formattedSummary,
            style = ScribbleFitTheme.typography.bodyMedium,
            color = ScribbleFitTheme.colors.midGray
        )
    }
}

@Composable
fun ExerciseStats(
    estimated1RM: String?,
    intensity: String?,
    improvement: String?,
    hasStats: Boolean,
    estimated1RMLabel: String,
    intensityLabel: String,
    modifier: Modifier = Modifier
) {
    if (!hasStats) return

    Column(
        modifier = modifier,
        verticalArrangement = Arrangement.spacedBy(ScribbleFitTheme.spacing.medium)
    ) {
        if (estimated1RM != null || intensity != null) {
            Row(horizontalArrangement = Arrangement.spacedBy(ScribbleFitTheme.spacing.medium)) {
                estimated1RM?.let { oneRm ->
                    StatCard(
                        label = estimated1RMLabel,
                        value = oneRm,
                        modifier = Modifier.weight(1f)
                    )
                }
                intensity?.let { intensityVal ->
                    StatCard(
                        label = intensityLabel,
                        value = intensityVal,
                        modifier = Modifier.weight(1f)
                    )
                }
            }
        }

        improvement?.let { improvementVal ->
            Row(
                verticalAlignment = Alignment.CenterVertically,
                horizontalArrangement = Arrangement.spacedBy(ScribbleFitTheme.spacing.extraSmall)
            ) {
                Icon(
                    imageVector = Icons.Rounded.History,
                    contentDescription = null,
                    tint = ScribbleFitTheme.colors.midGray,
                    modifier = Modifier.size(ScribbleFitTheme.spacing.smallLarger)
                )
                Text(
                    text = improvementVal.uppercase(),
                    style = ScribbleFitTheme.typography.labelMedium.copy(
                        fontWeight = FontWeight.SemiBold,
                        letterSpacing = ScribbleFitTheme.typography.labelMedium.letterSpacing
                    ),
                    color = ScribbleFitTheme.colors.midGray
                )
            }
        }
    }
}

@Composable
fun StatCard(label: String, value: String, modifier: Modifier = Modifier) {
    Box(
        modifier = modifier
            .clip(RoundedCornerShape(ScribbleFitTheme.shapes.small))
            .background(ScribbleFitTheme.colors.primary.copy(alpha = ScribbleFitTheme.Alphas.cardOverlay))
            .padding(ScribbleFitTheme.spacing.smallLarger)
    ) {
        Column(verticalArrangement = Arrangement.spacedBy(ScribbleFitTheme.spacing.extraSmall)) {
            Text(
                text = label.uppercase(),
                style = ScribbleFitTheme.typography.labelSmall,
                color = ScribbleFitTheme.colors.midGray,
                fontWeight = FontWeight.Bold,
                letterSpacing = ScribbleFitTheme.typography.labelSmall.letterSpacing
            )
            Text(
                text = value,
                style = ScribbleFitTheme.typography.titleMedium,
                fontWeight = FontWeight.Bold,
                color = ScribbleFitTheme.colors.primary
            )
        }
    }
}

File: apps/ios/ScribbleFit/LocalPackages/Core/CoreDesignSystem/Sources/CoreDesignSystem/ExerciseCardComponents.swift
import SwiftUI

public struct GlassCard<Content: View>: View {
    let onClick: (() -> Void)?
    let alpha: Double
    let content: Content

    public init(
        onClick: (() -> Void)? = nil,
        alpha: Double = 1.0,
        @ViewBuilder content: () -> Content
    ) {
        self.onClick = onClick
        self.alpha = alpha
        self.content = content()
    }

    public var body: some View {
        if let onClick {
            Button(action: onClick) {
                cardContent
            }
            .buttonStyle(.plain)
        } else {
            cardContent
        }
    }

    private var cardContent: some View {
        VStack(alignment: .leading, spacing: 16) {
            content
        }
        .padding(20)
        .scribbleGlass(cornerRadius: 16)
        .opacity(alpha)
    }
}

public struct ExerciseHeaderView: View {
    let name: String
    let formattedSummary: String
    let fontSize: CGFloat
    let kerning: CGFloat

    public init(
        name: String,
        formattedSummary: String,
        fontSize: CGFloat = 24,
        kerning: CGFloat = -0.5
    ) {
        self.name = name
        self.formattedSummary = formattedSummary
        self.fontSize = fontSize
        self.kerning = kerning
    }

    public var body: some View {
        VStack(alignment: .leading, spacing: 4) {
            Text(name)
                .font(.system(size: fontSize, weight: .bold))
                .kerning(kerning)
                .foregroundStyle(Color.scribblePrimary)

            Text(formattedSummary)
                .font(.scribbleBodyMedium)
                .foregroundStyle(Color.scribbleMidGray)
        }
    }
}

public struct ExerciseStatsView: View {
    let estimated1RM: String?
    let intensity: String?
    let improvement: String?

    public init(
        estimated1RM: String?,
        intensity: String?,
        improvement: String? = nil
    ) {
        self.estimated1RM = estimated1RM
        self.intensity = intensity
        self.improvement = improvement
    }

    public var hasStats: Bool {
        estimated1RM != nil || intensity != nil || improvement != nil
    }

    public var body: some View {
        if hasStats {
            VStack(alignment: .leading, spacing: 12) {
                if estimated1RM != nil || intensity != nil {
                    HStack(spacing: 12) {
                        if let oneRm = estimated1RM {
                            StatCardView(label: String(localized: "EST. 1RM"), value: oneRm)
                        }
                        if let intensityVal = intensity {
                            StatCardView(label: String(localized: "INTENSITY"), value: intensityVal)
                        }
                    }
                }

                if let improvementVal = improvement {
                    HStack(spacing: 4) {
                        Image(systemName: "clock.arrow.circlepath")
                            .font(.system(size: 12))
                            .foregroundStyle(Color.scribbleMidGray)

                        Text(improvementVal.uppercased())
                            .font(.scribbleLabelMedium)
                            .fontWeight(.semibold)
                            .kerning(0.5)
                            .foregroundStyle(Color.scribbleMidGray)
                    }
                }
            }
        }
    }
}

public struct StatCardView: View {
    let label: String
    let value: String

    public init(label: String, value: String) {
        self.label = label
        self.value = value
    }

    public var body: some View {
        VStack(alignment: .leading, spacing: 4) {
            Text(label)
                .font(.scribbleLabelMedium)
                .fontWeight(.bold)
                .kerning(1)
                .foregroundStyle(Color.scribbleMidGray)

            Text(value)
                .font(.scribbleTitleMedium)
                .fontWeight(.bold)
                .foregroundStyle(Color.scribblePrimary)
        }
        .frame(maxWidth: .infinity, alignment: .leading)
        .padding(12)
        .background(Color.scribblePrimary.opacity(0.03))
        .clipShape(RoundedRectangle(cornerRadius: 12))
    }
}
